saleor/graphql/core/types/taxes.py

```diff
@@ -0,0 +1,394 @@
+                        lines,
+                    ]
+                ).then(calculate_line_total_price)
+
+            return (
+                CheckoutByTokenLoader(info.context)
+                .load(root.checkout_id)
+                .then(with_checkout)
+            )
+        return root.base_unit_price * root.quantity
+
+
+class TaxableObjectDiscount(graphene.ObjectType):
+    name = graphene.String(description="The name of the discount.")
+    amount = graphene.Field(
+        Money, description="The amount of the discount.", required=True
+    )
+
+
+class TaxableObject(graphene.ObjectType):
+    source_object = graphene.Field(
+        TaxSourceObject,
+        required=True,
+        description="The source object related to this tax object.",
+    )
+    prices_entered_with_tax = graphene.Boolean(
+        required=True, description="Determine if prices contain entered tax."
+    )
+    currency = graphene.String(required=True, description="The currency of the object.")
+    shipping_price = graphene.Field(
+        Money, required=True, description="The price of shipping method."
+    )
+    address = graphene.Field(
+        "saleor.graphql.account.types.Address",
+        description="The address data.",
+    )
+    discounts = NonNullList(
+        TaxableObjectDiscount, description="List of discounts.", required=True
+    )
+    lines = NonNullList(
+        TaxableObjectLine,
+        description="List of lines assigned to the object.",
+        required=True,
+    )
+
+    class Meta:
+        description = "Taxable object."
+
+    def resolve_address(root: Union[Checkout, Order], info):
+        address_id = root.shipping_address_id or root.billing_address_id
+        if not address_id:
+            return None
+        return AddressByIdLoader(info.context).load(address_id)
+
+    @staticmethod
+    def resolve_source_object(root: Union[Checkout, Order], info):
+        return root
+
+    @staticmethod
+    def resolve_prices_entered_with_tax(root: Union[Checkout, Order], info):
+        return include_taxes_in_prices()
+
+    @staticmethod
+    def resolve_currency(root: Union[Checkout, Order], info):
+        return root.currency
+
+    @staticmethod
+    def resolve_shipping_price(root: Union[Checkout, Order], info):
+        if isinstance(root, Checkout):
+
+            def calculate_shipping_price(data):
+                checkout_info, lines = data
+                is_shipping_voucher = (
+                    checkout_info.voucher.type == VoucherType.SHIPPING
+                    if checkout_info.voucher
+                    else False
+                )
+                price = base_calculations.base_checkout_delivery_price(
+                    checkout_info, lines
+                )
+                if is_shipping_voucher:
+                    price.amount = max(
+                        price.amount - checkout_info.checkout.discount_amount,
+                        Decimal("0.0"),
+                    )
```

Comment:
Shouldn't this logic be encapsulated inside the `base_checkout_delivery_price` function? I'd just like to call this function and get the right shipping price. What do you think @fowczarek? And if there is a reason to add this condition here, maybe we should add a comment.

Reply:
I used the old webhook flow introduced in tax webhook to determine the way of building the payload. 
AFAIK the calculation of shipping price for tax webhook is slightly different than for our basic resolvers. We want to build a payload, where the shipping price already takes into account a shipping voucher. I could add it to base_calculation file, but this could be also misleading for someone who wants to get a shipping_price for our basic calculation flow.

Reply:
The solution proposed by me is described in the internal task https://app.clickup.com/t/2549495/SALEOR-7786


> In order, we store only the actual shipping price as TaxedMoeney. A tax app or plugin could overrate this price. We should keep the base shipping price(As Money value) when creating orders, and use this value in all feature tax calculations. 
> We have a similar solution for line price.

Reply:
OK so for now we keep it and resolve as part of the task you linked, is that right @fowczarek ?

Reply:
Yes.

Reply:
IMO, yes.